crates/test/mock-consensus/src/builder/init_chain.rs

```diff
@@ -0,0 +1,94 @@
+use {
+    super::*,
+    anyhow::{anyhow, bail},
+    std::time,
+    tap::TapFallible,
+    tendermint::{
+        block,
+        consensus::{
+            self,
+            params::{AbciParams, ValidatorParams, VersionParams},
+        },
+        evidence,
+        v0_37::abci::{ConsensusRequest, ConsensusResponse},
+    },
+    tower::{BoxError, Service, ServiceExt},
+    tracing::{debug, error},
+};
+
+impl Builder {
+    /// Consumes this builder, using the provided consensus service.
+    pub async fn init_chain<C>(self, mut consensus: C) -> Result<TestNode<C>, anyhow::Error>
+    where
+        C: Service<ConsensusRequest, Response = ConsensusResponse, Error = BoxError>
+            + Send
+            + Clone
+            + 'static,
+        C::Future: Send + 'static,
+        C::Error: Sized,
+    {
+        use tendermint::v0_37::abci::response;
+
+        let request = Self::init_chain_request();
+        let service = consensus
+            .ready()
+            .await
+            .tap_err(|error| error!(?error, "failed waiting for consensus service"))
+            .map_err(|_| anyhow!("failed waiting for consensus service"))?;
+
+        let response::InitChain { app_hash, .. } = match service
+            .call(request)
+            .await
+            .tap_ok(|resp| debug!(?resp, "received response from consensus service"))
+            .tap_err(|error| error!(?error, "consensus service returned error"))
+            .map_err(|_| anyhow!("consensus service returned error"))?
+        {
+            ConsensusResponse::InitChain(resp) => resp,
+            response => {
+                error!(?response, "unexpected InitChain response");
+                bail!("unexpected InitChain response");
+            }
+        };
+
+        Ok(TestNode {
+            consensus,
+            last_app_hash: app_hash.as_bytes().to_owned(),
+        })
+    }
+
+    fn init_chain_request() -> ConsensusRequest {
```

Comment:
probably this will want to be `&self` in the future so we can read stuff off the builder (assuming it's not like that for now to suppress unused var warnings)

Reply:
:heavy_plus_sign: i agree, it's just like this for now until i do need those fields. lazy evaluation, so to speak